specification/under_development/profiles/Adaptive_Profile.md

```diff
@@ -0,0 +1,1028 @@
+reflect the program intent. This includes being able to define and customize the
+program output. The base profile and adaptive profile specifications hence
+require explicitly expressing which values/measurements are returned by the
+program and in which order. How to express this is defined in the section on
+[output recording](#output-recording).
+
+A suitable output schema can be generated in a post-processing step after the
+computation on the quantum processor itself has finished; customization of the
+program output hence does not require support on the QPU itself.
+
+The defined [output schemas](../output_schemas/) provide different options for
+how a backend may express the computed value(s). The exact schema can be freely
+chosen by the backend and is identified by a string label in the produced
+schema. Each output schema contains sufficient information to allow quantum
+programming frameworks to generate a user-friendly presentation of the returned
+values in the requested order, such as, e.g., a histogram of all results when
+running the program multiple times.
+
+### Bullet 4: Conditional Forward branching
+
+The adaptive profile can allow for arbitrary forward branching with the
+restriction being that branch instructions cannot express programs that do not
+terminate. By default, support for backwards branching is not a requirement,
+though a backend can opt into **Bullet 11** to support backwards branching.
+In the case that **Bullet 11** is not opted into, then a simple static analysis
+can check for cycles in the control flow graph and reject any program with a
+cycle as invalid. Depending on the vector of capabilities defined by an adaptive
+profile program (like when **Bullet 11** is opted into), proving non-termination
+with a static analysis may be impossible due to the capabilities forming a
+Turing-complete subset of LLVM. Therefore, it is a rule that a *valid* adaptive
+profile program should not contain non-terminating loops and that a frontend
+generating a program with a non-terminating loop is generating a program that is
+not compliant with the adaptive profile specification. However, since there is
+no static analysis that can prove termination then it is up to backends to enforce
+termination guarantees via a means of their choosing (for example a watchdog process
+with a timeout that will kill an executing adaptive profile program if it takes
+too much time). Different forms of conditional logic can be supported in the
+profile based on optional features, but at minimum, the simple `br` branch
+instruction must be supported.
+
+### Bullet 5: Mid-circuit measurement
+
+The adaptive profile allows for mid-circuit measurement to be expressed in
+programs by removing restrictions on a measurement operation (we will use
+measurement in Z represented by `llvm __quantum__qis__mz__body` in this
+document, but other forms of measurement can also apply) that state that
```

Comment:
```suggestion
document, but other forms of measurement can be substituted) that state that
```
Note that there is a finer decision point here regarding whether a backend that only supports joint measurements is profile compliant. I.e. imo the adaptive spec adds capability requirements for how *whatever* measurements that are supported by a backend can be used, but doesn't require *specific* measurements to be possible. I advocate for this being the case. Hence, my phrasings above carefully avoid stating that it must be possible to measure any *single* qubit, and updated the phrasing here to reflect the fact that we use mz as a placeholder for whatever measurements the backend supports.